src/internal.rs

```diff
@@ -155,27 +158,26 @@ impl BitDepthDependentType for Grain {
 #[repr(C)]
 pub(crate) struct TaskThreadData_delayed_fg {
     pub exec: c_int,
-    pub cond: pthread_cond_t,
     pub in_0: *const Rav1dPicture,
     pub out: *mut Rav1dPicture,
     pub type_0: TaskType,
-    pub progress: [atomic_int; 2], /* [0]=started, [1]=completed */
     pub grain: BitDepthUnion<Grain>,
 }
 
 #[repr(C)]
 pub(crate) struct TaskThreadData {
-    pub lock: pthread_mutex_t,
-    pub cond: pthread_cond_t,
+    pub lock: Mutex<TaskThreadData_delayed_fg>,
```

Comment:
```suggestion
    pub delayed_fg: Mutex<TaskThreadData_delayed_fg>,
```

And for the `ttd_locked`-like variables, can you just name them `delayed_fg`?